OJS.Workers.SubmissionProcessors/ExecutionTypeFilters/SubmissionFilteringServiceBase.cs

```diff
@@ -67,8 +68,17 @@ public bool CanProcessSubmission(IOjsSubmission submission, ISubmissionWorker su
             }
 
             this.logger.Error($"Submission with Id: {submission.Id}, cannot be processed. Reason: {reason} ");
+            if (!canProcessSubmissionInternal)
+            {
+                return false;
+            }
+            else
+            {
+                submission.CompilerComment =
+                    "The submission cannot be handled due to unsupported submission type. Please contact an Administrator.";
 
-            return false;
+                throw new Exception($"Submission with Id: {submission.Id}, cannot be processed. Reason: {reason} ");
+            }
```

Comment:
It's not very intuitive to return boolean if it's `true` and `false` only in one single case and throw exception in every other case.
Exception throwing is something we should avoid for better readability, maintainability etc.
Better approach would be to return Enum from this method. e.g `WorkerState` that could be some of the following:
```
Ready,
Unresponsive,
Unhealthy,
DisabledStrategy,
DisabledCompilerType,
```

Based on these the SubmissionProcessor can attach different messages to the submission, without the need to catch exceptions.

Reply:
Done.